sklearn/ensemble/iforest.py

```diff
@@ -61,6 +67,12 @@ class IsolationForest(BaseBagging):
             - If int, then draw `max_features` features.
             - If float, then draw `max_features * X.shape[1]` features.
 
+    max_depth : int or float, optional (default="auto")
+        The max_depth of the tree.
+            - If int, then maximum depth is `max_depth`.
+            - If float, then maximum depth is  `max_depth * max_samples`.
```

Comment:
It is weird to have a linear relationship between max_depth and max_samples, as there is a log_2 between them in average when the tree is fully grown. Moreover, it is not consistent with other tree-based estimators. I think we should add a 'None' option instead, for fully expanded trees.